pkg/manager/internal.go

```diff
@@ -311,12 +334,80 @@ func (cm *controllerManager) addMetricsServer() error {
 
 	return cm.add(&server{
 		Kind:     "metrics",
-		Log:      cm.logger.WithValues("path", defaultMetricsEndpoint),
+		Log:      log,
 		Server:   srv,
 		Listener: cm.metricsListener,
 	})
 }
 
+func withAuthenticationAndAuthorization(log logr.Logger, metricsKubeClient *kubernetes.Clientset, handler http.Handler) (http.Handler, error) {
```

Comment:
I think the implementation is pretty straightforward because of the functionality we can re-use from k8s.io/apiserver.

The downside is unfortunately a lot of new indirect dependencies. Not sure if this is acceptable.
If not I can explore how much effort it would be to implement this functionality ourselves, but I think it's considerable.

Reply:
Found a way that folks won't get the dependency if they don't use the filter